### PR TITLE
Josiahsiegel/data factory proto

### PR DESCRIPTION
**Pull request by JosiahSiegel**
This PR brings in the **data factory** and all the basic connections that go with it.
If the data factory becomes part of the final setup, then we can also bring additional data factory components, but this will get us started.

TF has also been cleaned up a bit and optimized where possible.

### CODE REVIEW

File: operations/app/terraform/modules/log_analytics_workspace/main.tf [hunk 1]
```diff
@@ -1,143 +1,44 @@
...
-
-resource "azurerm_monitor_diagnostic_setting" "vnet_diag" {
-  name                       = "vnet_diag"
-  target_resource_id         = var.cdc_managed_vnet_id
-  log_analytics_workspace_id = azurerm_log_analytics_workspace.pdi.id
-
-  log {
-    category = "VMProtectionAlerts"
-    enabled  = true
+  dynamic "log" {
```
Comment (cwinters-usds) [inline operations/app/terraform/modules/log_analytics_workspace/main.tf:15]: TIL :raised_hands:

File: operations/app/terraform/vars/test/variables.tf [hunk 2]
```diff
@@ -81,5 +81,6 @@ variable "use_cdc_managed_vnet" {
 }
 
 variable "terraform_caller_ip_address" {
-  default = "162.224.209.174"
+  type    = list(string)
+  default = ["162.224.209.174/32","73.173.186.141/32","24.163.118.70/32"]
```
Comment (cwinters-usds) [inline operations/app/terraform/vars/test/variables.tf:85]: are these additional addresses from GitHub or...?

Comment (JosiahSiegel) [reply operations/app/terraform/vars/test/variables.tf:85]: 73 is from @NatHillardUSDS and 24 is myself

**Review by cwinters-usds — APPROVED** [review]
Thank you -- this is a lot of wiring!

**Pull request merged**